cmd/mitm/main.go

```diff
@@ -0,0 +1,170 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+	"os/exec"
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/deref/exo/internal/core/api"
+	"github.com/deref/exo/internal/core/client"
```

Comment:
Since this client is internal to the exo package the `mitm` binary needs to be part of the same package. That means that we need to set up a build environment for the entirety of exo in order to be able to build the relevant binary. This makes building the appropriate docker image a great deal slower than it really needs to be. I'd propose moving the client out of the `internal` directory so that we can just build what we need.